### PR TITLE
option to set the address prometheus is on

### PR DESCRIPTION
**Pull request by cglewis**

### CODE REVIEW

**Comment by lantz** [comment]
@cglewis Could you explain the use case for this or the bug it fixes? The original spec was for the agent to run on the same machine as FAUCET (otherwise you may have to jump through hoops to make sure that the file system and pid space are shared, for example.)

**Comment by cglewis** [comment]
So the issue I was having with it being `localhost` was when running Faucet in a Docker container either manually or using the docker-compose, the default `docker-compose.yaml` for Faucet doesn't expose 9302 so `localhost` is only within the scope of inside the container, not the host it's running on.  I see your point about the pid space, though it wouldn't be an issue if using the `FAUCET_CONFIG_STAT_RELOAD` instead of `HUP` I think? And a shared filesystem with Docker volumes is either already being done or easy enough to do.

Basically I want to be able to run Faucet in a container and FaucetAgent in a different container on the same host, I saw this as a stepping stone to be able to get there.

**Comment by lantz** [comment]
I was afraid you'd say something like that; I'm not entirely enthusiastic about decoupling the agent from FAUCET, as it seems more likely to introduce unexpected behavior and errors (such as having more than one agent trying to control FAUCET at the same time, having an agent connected to the wrong FAUCET, or writing the wrong config file) but we can do it.

**Comment by gizmoguy** [comment]
Sounds like something we might want to discuss on the dev call tomorrow?

**Comment by lantz** [comment]
After discussing it today, I think this is fine and we can move forward on it; if you want to add a simple test (and perhaps allow specifying the whole URL including `https` if desired?) then think we should merge it in.

**Comment by cglewis** [comment]
@lantz tests have been added. I'll have to update this after #24 is merged in though, as it will have conflicts.

File: faucetagent.py [hunk 1]
```diff
@@ -332,7 +335,10 @@ def parse():
     arg('--key', required=True, help='private key file')
     arg('--gnmiport', type=int, default=10161, help='gNMI port (10161)')
     arg('--configfile', required=True, help='FAUCET config file')
-    arg('--promport',
+    arg('--prom_addr',
```
Comment (lantz) [inline faucetagent.py]: I think we should be consistent with underscores in command line options; I find them inconvenient to type so my preference is to not use them in command line options. PEP8 (annoyingly) recommends their use in Python code and that also seems to be the FAUCET style, so that's why the internal variable names are different. We could change them internally for consistency (though we might potentially have to silence some pylint complaints.)

File: faucetagent.py [hunk 2]
```diff
@@ -332,7 +335,10 @@ def parse():
...
+        default='http://localhost',
+        help='FAUCET prometheus address (http://localhost)')
+    arg('--prom_port',
```
Comment (lantz) [inline faucetagent.py]: We had `--promport` before so I think we don't need to add an underscore.

**Comment by cglewis** [comment]
Ok, changed.
